shuffle/cli/src/main.rs

```diff
@@ -162,10 +176,10 @@ fn normalized_key_path(diem_root_key_path: Option<PathBuf>) -> Result<PathBuf> {
     }
 }
 
-fn normalized_network(network: &str) -> Result<Url> {
+fn normalized_network(network: &str) -> Result<Url, ParseError> {
     match Url::parse(network) {
         Ok(_res) => Ok(Url::parse(network)?),
-        Err(_e) => Ok(Url::parse(("http://".to_owned() + network).as_str())?),
+        Err(parse_error) => Err(parse_error),
     }
```

Comment:
You're still doing a double `Url::parse` unnecessarily. The way you currently have this, you could replace the whole block with simply

```rust
Url::parse(network)
```

You're not normalizing anything here anymore, which is wrong. Why is `network: String` not optional?

Reply:
Whatever you do to `network` for the Transaction cmd, you will want to be consistent with the `network` for the Console cmd.